core/ledger/kvledger/txmgmt/privacyenabledstate/snapshot.go

```diff
@@ -65,30 +45,61 @@ func (s *DB) ExportPubStateAndPvtStateHashes(dir string, newHashFunc snapshot.Ne
 		}
 		switch {
 		case isHashedDataNs(compositeKey.Namespace):
+			if pvtStateHashesWriter == nil { // encountered first time the pvt state hash element
+				pvtStateHashesWriter, err = newSnapshotWriter(
+					filepath.Join(dir, pvtStateHashesFileName),
+					filepath.Join(dir, pvtStateHashesMetadataFileName),
+					dbValueFormat,
+					newHashFunc,
+				)
+				if err != nil {
+					return nil, err
+				}
+				defer pvtStateHashesWriter.close()
+			}
 			if err := pvtStateHashesWriter.addData(compositeKey, dbValue); err != nil {
 				return nil, err
 			}
 		default:
+			if pubStateWriter == nil { // encountered first time the pub state element
+				pubStateWriter, err = newSnapshotWriter(
+					filepath.Join(dir, pubStateDataFileName),
+					filepath.Join(dir, pubStateMetadataFileName),
+					dbValueFormat,
+					newHashFunc,
+				)
+				if err != nil {
+					return nil, err
+				}
+				defer pubStateWriter.close()
+			}
 			if err := pubStateWriter.addData(compositeKey, dbValue); err != nil {
 				return nil, err
 			}
 		}
 	}
-	pubStateDataHash, pubStateMetadataHash, err := pubStateWriter.done()
-	if err != nil {
-		return nil, err
+
```

Comment:
Nit. 

Could return early here too if both writers are nil (just for the consistency with other export APIs).

Reply:
I'll leave this as is. You typically return early to shorten an `if` block. Not applicable here.